### DISCUSSION TITLE
Need CAN support for ESP32 S2

### DISCUSSION THREAD

**Discussion by PriyadharshiniMohan** [zephyrproject-rtos/zephyr#49561] [Q&A] [answered]
Hi
I am working on Zephyr platform. Currently I have support for CAN module on ESP32. The same code cannot be built for ESP32 S2. There are issues in Zephyr dependency for ESP32 S2. Also I need to know the where the pin assignment has to be made for CAN.
Thanks
Priya

**Comment by rftafas** [comment]
There is already a merged PR with TWAI support on ESP32. Can you check if it is enough?

Please, report any issues you find so we can put that into our to-do list. Same with questions or need for how-to's.

**Comment by henrikbrixandersen** [comment, marked as answer]
PR https://github.com/zephyrproject-rtos/zephyr/pull/51100 by @martinjaeger was merged 16 days ago. It should provide the basic support for TWAI on both ESP32-S2 and ESP32-C3.

An example of how to enable the TWAI peripheral and set up the pins in a board definition  can be found here:
- https://github.com/zephyrproject-rtos/zephyr/blob/35ec706d82a562dbefe2c2744d3e4e9406ee92b7/boards/xtensa/olimex_esp32_evb/olimex_esp32_evb.dts#L96
- https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/xtensa/olimex_esp32_evb/olimex_esp32_evb-pinctrl.dtsi#L47
